lib/ddtrace/monkey.rb

```diff
@@ -18,51 +19,30 @@
 module Datadog
   # Monkey is used for monkey-patching 3rd party libs.
   module Monkey
-    @patched = []
-    @autopatch_modules = {
-      elasticsearch: true,
-      http: true,
-      redis: true,
-      grape: true,
-      faraday: true,
-      aws: true,
-      sucker_punch: true,
-      mongo: true,
-      dalli: true,
-      resque: true,
-      active_record: false
-    }
     # Patchers should expose 2 methods:
     # - patch, which applies our patch if needed. Should be idempotent,
     #   can be call twice but should just do nothing the second time.
     # - patched?, which returns true if the module has been succesfully
     #   patched (patching might have failed if requirements were not here)
-    @patchers = { elasticsearch: Datadog::Contrib::Elasticsearch::Patcher,
-                  http: Datadog::Contrib::HTTP::Patcher,
-                  redis: Datadog::Contrib::Redis::Patcher,
-                  grape: Datadog::Contrib::Grape::Patcher,
-                  faraday: Datadog::Contrib::Faraday::Patcher,
-                  aws: Datadog::Contrib::Aws::Patcher,
-                  sucker_punch: Datadog::Contrib::SuckerPunch::Patcher,
-                  mongo: Datadog::Contrib::MongoDB::Patcher,
-                  dalli: Datadog::Contrib::Dalli::Patcher,
-                  resque: Datadog::Contrib::Resque::Patcher,
-                  active_record: Datadog::Contrib::ActiveRecord::Patcher }
+
     @mutex = Mutex.new
+    @registry = Datadog.registry
```

Comment:
should we really keep `Datadog.registry` as a reference + `attr_accessor`? what is the use case to not use directly `Datadog.registry`?

Reply:
Basically by injecting the dependency we can test everything in isolation. Right now testing the `Monkey` class is a nightmare, because every time we create a new integration we have to update 10 lines of test in the `Monkey`. In my opinion `Monkey` tests should be agnostic of the existing contributions. `Monkey` should be responsible for calling `#patch` in contribution modules, that's it. By having that, we can write such tests.

Reply:
👍